Frameworks/UIKit/UIViewController.mm

```diff
@@ -1222,9 +1230,9 @@ - (void)dismissViewControllerAnimated:(BOOL)animated completion:(void (^)(void))
     [curController retain];
     [curController autorelease];
 
-    BOOL popoverPresented = [curController _popoverManagesPresentation];
+    BOOL realPopoverPresented = [[curController popoverPresentationController] _isManagingPresentation];
```

Comment:
@jaredhms I've added a comment. Regarding your message (which I think was deleted) about whether to simplify the implementation by always showing a popover on phones/non-tablets, I'm initially supportive though would prefer a separate pull. The behaviour has changed in iOS 9 anyway to allow popovers on phones in "non-horizontally compact" environments. There would also be an easy workaround for users that don't want the popover - just use a different modal presentation style. Simplification wise, this would allow the popover presentation controller setup to be moved out of _addToTop and into presentViewController:. I'm also not sure whether the UIPopoverPresentationControllerDelegate is being handled correctly for non actual popover presentations right now (I'm not sure if calling dismissViewController: should fire the didDismiss delegate - would have to be checked) - so it probably would be an improvement to just always show a popover.

We'd have to consider the behaviour of UIModalPresentationFormSheet on non-tablets as well. There I'm thinking that the current behaviour to display as full screen might still be best (since the default hardcoded dimensions of the FormSheet modal are inappropriate for phones) - though on the other hand these are resizable via preferredContentSize.

Note however, there are a few framework issues I need to file that have come up regarding popovers though the problems are unrelated to this pull (just a heads up).